src/router.ts

```diff
@@ -263,9 +266,10 @@ export class DocumentRoute extends Route {
     return `[DocumentRoute: ${this.doc}]`;
   }
 
-  async build(): Promise<string> {
+  async build(req?: express.Request): Promise<string> {
```

Comment:
Let's make this use an interface `BuildOptions` and have a single property `req?` on it. I think that'll be more consistent with our usage elsewhere.